activation/activation.go

```diff
@@ -386,9 +386,15 @@ func (b *Builder) loop(ctx context.Context) {
 
 			switch {
 			case errors.Is(err, ErrATXChallengeExpired):
-				b.log.WithContext(ctx).Debug("discarding challenge")
+				b.log.WithContext(ctx).Debug("retrying with new challenge after waiting for a layer")
```

Comment:
does it make sense to wait for time-based instead of layer based? atx can arrive any time.

Reply:
I think that it would be most robust to wait until we receive a positioning ATX with PubLayerID = current challenge's PositioninigATX.PubLayerID + 1. 

For this, the `AtxHandler` would need a new method like `AwaitAtxWithPublayerID(lid types.LayerID)`.

Reply:
> I think that it would be most robust to wait until we receive a positioning ATX with PubLayerID = current challenge's PositioninigATX.PubLayerID + 1.

There might not be an ATX in that exact layer, imo t would need to be an ATX with `PubLayerID` >= `currentlayer.GetEpoch().FirstLayer()`. But wouldn't this be the same as `RegisterForATXSynced()`?

The reason this happens is because the node is not in sync with the network but believes it is. This is also why restarting the node fixes the issue. A proper fix imo would update `RegisterForATXSynced()` to not return a closed channel if the node is not in sync with the network, but this would be a much bigger change.

The "good enough for now" fix is to just wait between attempts to give the node time and resources to sync new ATXs to hopefully have one in the current epoch on the next try.

I choose to wait for the next layer because it scales with the network. It would only wait 15 seconds in system tests and 5 minutes on mainnet. A fixed time would probably be too long for system tests or not long enough for mainnet and this imo shouldn't be configurable.

Reply:
not sure why `RegisterForATXSynced()` matters. atx synced only happens once in a node's lifetime. once the first time is done, every call afterwards will return immediately.

Reply:
I know, but this error happens when a node is not ATX synced (meaning it hasn't seen any ATXs that were published in the current epoch) and still returns a closed channel via `RegisterForATXSynced()`.

A node can be ATX synced and then lose this state - by being offline for some time or hibernating, or other causes.